numba_cuda/numba/cuda/tests/cudapy/test_cooperative_groups.py

```diff
@@ -149,6 +157,32 @@ def test_max_cooperative_grid_blocks(self):
         self.assertEqual(blocks1d, blocks2d)
         self.assertEqual(blocks1d, blocks3d)
 
+    @skip_unless_cc_60
+    def test_external_coorperative_func(self):
+        sig = signature(
+            CPointer(int32),
+        )
+        cta_barrier = cuda.declare_device(
+            "cta_barrier", sig=sig, use_cooperative=True
+        )
+
+        cudapy_test_path = os.path.dirname(__file__)
+        tests_path = os.path.dirname(cudapy_test_path)
+        data_path = os.path.join(tests_path, "data")
+        src = os.path.join(data_path, "cta_barrier.cu")
+
+        @cuda.jit(link=[src])
```

Comment:
Would it be more representative of the way we hope APIs will be used to put the `link` data in the `cta_barrier` declaration above?